core/src/main/java/com/datastax/oss/driver/internal/core/loadbalancing/DefaultLoadBalancingPolicy.java

```diff
@@ -96,14 +100,39 @@ public class DefaultLoadBalancingPolicy extends BasicLoadBalancingPolicy impleme
   private static final int MAX_IN_FLIGHT_THRESHOLD = 10;
   private static final long RESPONSE_COUNT_RESET_INTERVAL_NANOS = MILLISECONDS.toNanos(200);
 
-  protected final Map<Node, AtomicLongArray> responseTimes = new ConcurrentHashMap<>();
+  protected final LoadingCache<Node, AtomicLongArray> responseTimes;
   protected final Map<Node, Long> upTimes = new ConcurrentHashMap<>();
   private final boolean avoidSlowReplicas;
 
   public DefaultLoadBalancingPolicy(@NonNull DriverContext context, @NonNull String profileName) {
     super(context, profileName);
     this.avoidSlowReplicas =
         profile.getBoolean(DefaultDriverOption.LOAD_BALANCING_POLICY_SLOW_AVOIDANCE, true);
+    CacheLoader<Node, AtomicLongArray> cacheLoader =
+        new CacheLoader<Node, AtomicLongArray>() {
+          @Override
+          public AtomicLongArray load(Node key) throws Exception {
+            // The array stores at most two timestamps, since we don't need more;
+            // the first one is always the least recent one, and hence the one to inspect.
+            long now = nanoTime();
+            AtomicLongArray array =
+                responseTimes.asMap().containsKey(key) ? responseTimes.get(key) : null;
+            if (array == null) {
+              array = new AtomicLongArray(1);
+              array.set(0, now);
+            } else if (array.length() == 1) {
+              long previous = array.get(0);
+              array = new AtomicLongArray(2);
+              array.set(0, previous);
+              array.set(1, now);
```

Comment:
nit: construction + sets could be replaced with `new AtomicLongArray(new long[] {previous, now})` unless there is a good reason that performs poorly

Reply:
it currently becomes 
```java
            AtomicLongArray array = responseTimes.getIfPresent(key);
            if (array == null) {
              array = new AtomicLongArray(new long[] {-1, now});
            } else {
              array = new AtomicLongArray(new long[] {array.get(1), now});
            }
```
Do you think this implementation is proper?

Reply:
the [documentation](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#nanoTime--) claims nanotime can be negative so I don't think you can use `-1` to represent a special value

Reply:
thx, reverted